src/TopDownProteomics/ProForma/TopPicProformaParser.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace TopDownProteomics.ProForma
+{
+    /// <summary>
+    /// A parser for TopPic strings into a ProformaTerm <see cref="ProFormaTerm"/>
+    /// </summary>
+    public class TopPicProformaParser
+    {
+        IDictionary<string, ProFormaDescriptor>? _modLookup = null;
+
+        #region Regex strings
+        Regex _modRx = new Regex(@"\(([A-Z]{1,})\)(\[[\w_+-.]+\])+");
+        Regex _numberRx = new Regex(@"(-?\+?[0-9]+.[0-9]+)");
+        Regex _terminalAaRx = new Regex(@"\P{N}(\.)\P{N}??|\P{N}??(\.)\P{N}");
+        #endregion
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TopPicProformaParser"/> class.
+        /// </summary>
+        public TopPicProformaParser() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TopPicProformaParser"/> class.
+        /// </summary>
+        /// <param name="modFile">The mod.txt file for mapping modifications.</param>
+        public TopPicProformaParser(string modFile)
+        {
+            _modLookup = ParseModFile(modFile);
+        }
+
+        /// <summary>
+        /// Gets the proforma term.
+        /// </summary>
+        /// <param name="sequence">The sequence.</param>
+        /// <returns></returns>
+        public ProFormaTerm ParseTopPicString(string sequence)
+        {
+            //first remove terminaltags if there!
+            sequence = RemoveTerminalAAs(sequence);
+            var ptms = FindPTMs(sequence);
```

Comment:
tuples are fine to use, but this line and the following aren't very readable. Switch to something like:
`var (tags, nTerm, cTerm) = FindPTMs(sequence);`